hw/ssi/spi_gpio.c

```diff
@@ -25,22 +25,89 @@
 #include "hw/ssi/spi_gpio.h"
 #include "hw/irq.h"
 
-static void cs_handler(void *opaque, int n, int level)
+#define SPI_CPHA	BIT(0)	/* clock phase (1 = SPI_CLOCK_PHASE_SECOND) */
+#define SPI_CPOL	BIT(1)	/* clock polarity (1 = SPI_POLARITY_HIGH) */
+
+static void do_leading_edge(SpiGpioState *s)
+{
+    int cpha = !!(s->mode & SPI_CPHA);
+
+    if (cpha) {
+       s->output_byte |= (s->mosi << s->clk_counter);
+    } else {
+       s->input_byte <<= 1;
+       s->input_byte |= s->mosi ? 1 : 0;
+    }
```

Comment:
```suggestion
    if (s->CPHA) {
       object_property_set_bool(OBJECT(s->aspeed_gpio),
                                "gpioX5", !!(s->output_byte & 0x80), NULL);
    } else {
       s->input_byte |= s->mosi ? 1 : 0;
    }
```
let's get the CPHA calculated in realize and use here.
and put the shift of input_byte and output_byte in trailing edge.